src/main/java/com/xeeva/catalog/RequisitioningPage.java

```diff
@@ -1,118 +1,458 @@
+				if(localAddToCartItemLinks.size()>0){
+					for(WebElement linkAddToCartItem :localAddToCartItemLinks){
+						TestReporter.log("Click on 'Add-To-Cart' link.");
+						linkAddToCartItem.click();
+						break;
+					}
+				}
+				break;
+			}
+		}else if(getLocalItemsCount.contains("0")){
+			TestReporter.log(" 'No Records Found !!' in Local Items tab. "
+					+ "Searching for Global items.");
+
+		}else if(!getGlobalItemsCount.contains("0")){
+			TestReporter.log("Click on Global-Items tab.");
+			GlobalItemsTab globalItem = new GlobalItemsTab(driver);
+			globalItem.click_GlobalItemsTab();
+
+			String itemNumber = null;
+			List<WebElement> globalItems = globalItemsGrid;
+			for(WebElement inputItem : globalItems){
+				itemNumber = inputItem.getText();
+				TestReporter.log("Item number is: "+itemNumber);
+				//Select the UOM option from the listbox.
+				List<WebElement> listboxUOMs = driver.findElements(By.xpath("//select[@class='textFieldList']"));
+				new Select(listboxUOMs.get(0)).selectByVisibleText(strUOMValue);
+				//Click on Add-to-cart link.
+				List<WebElement> globalAddToCartItemLinks = lstAddToCartItemsGrid;
+				if(globalAddToCartItemLinks.size()>0){
+					for(WebElement linkAddToCartItem :globalAddToCartItemLinks){
+						TestReporter.log("Click on 'Add-To-Cart' link.");
+						linkAddToCartItem.click();
+						break;
+					}
+				}
+				break;
+			}				
+		}else if(getGlobalItemsCount.contains("0")){
+			//If there are no records in both local & global items, then test should fail.
+			TestReporter.assertFalse(getGlobalItemsCount.contains("0"), " 'No Records Found !!' from Local & Global Items tab.");
+			
+		}
+
+	}
+
+
+	/**
+	 * @summary Method to click on FavoriteItem Link 
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void click_FavoriteItem(){
+		pageLoaded();
+		lnkShowFavItems.syncVisible(5, false);
+		TestReporter.assertTrue(lnkShowFavItems.isDisplayed(), "Fav Icon element is Displaying!!");
+		lnkShowFavItems.click();
+		Sleeper.sleep(2000);
+	}
+	
+	
+	  
+	  /**
+	   * @summary: Method to get the Item Number from catalog table.
+	   * @author praveen namburi,@version: Created 13-09-2016
+	   * @return getItemNumber
+	   */
+	  public String  getItemNumberFromCatalog(){
+		  Sleeper.sleep(2000);
+		  String getItemNumber = null;
+		  String getLocalItemsCount = lblLocalItems.getText();
+		  TestReporter.log("Local-Items Count is: "+ getLocalItemsCount);
+		  
+		  if(!getLocalItemsCount.contains("0")){
+				TestReporter.log("Click on Local-Items tab.");
+				LocalItemsTab localItem = new LocalItemsTab(driver);
+				localItem.click_localItemsTab();
+				List<WebElement> localItems = localItemsGrid;
+				for(WebElement inputItem : localItems){
+					String itemNumber = inputItem.getText().trim();
+					getItemNumber = itemNumber;
+					break;
+				}
+		  }else if(getLocalItemsCount.contains("0")){
+				TestReporter.log(" 'No Records Found !!' in Local Items tab. ");

```

Comment:
Line 450 - does this need to be an assertion statement so that the test fails if no records are found?